File: py/3160.py
def main():
    friends = input().split()
    new_friends = input().split()
    operation = input()
    result = []

    if operation == 'nao':
        print(*friends, *new_friends)
    
    else:
        for i in friends:
            if i == operation:
                result.extend(new_friends)

            result.append(i)

        print(*result)


main()

File: cpp/3160.cpp
#include <iostream>
#include <string>
#include <sstream>

using namespace std;

int main()
{
    string friends, new_friends, operation, word, result = "";
    getline(cin, friends);
    getline(cin, new_friends);
    getline(cin, operation);

    if (operation == "nao") {
        cout << friends << ' ' << new_friends << '\n';
    } else {
        stringstream ss(friends);

        while (ss >> word) {
            if (word == operation) {
                result += new_friends + ' ';
            }

            result += word + ' ';
        }

        result.pop_back();
        cout << result << '\n';
    }

    return 0;
}